File: sample/src/com/devspark/sidenavigation/sample/MainActivity.java
/*
 * Copyright (C) 2012 Evgeny Shishkin
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.devspark.sidenavigation.sample;

import android.content.Intent;
import android.os.Bundle;
import android.widget.ImageView;

import com.actionbarsherlock.app.SherlockActivity;
import com.actionbarsherlock.view.MenuItem;
import com.devspark.sidenavigation.ISideNavigationCallback;
import com.devspark.sidenavigation.SideNavigationView;

/**
 * 
 * @author e.shishkin
 * 
 */
public class MainActivity extends SherlockActivity implements
        ISideNavigationCallback {

    public static final String EXTRA_TITLE = "com.devspark.sidenavigation.sample.extra.MTGOBJECT";
    public static final String EXTRA_RESOURCE_ID = "com.devspark.sidenavigation.sample.extra.RESOURCE_ID";

    private ImageView icon;
    private SideNavigationView sideNavigationView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_main);
        icon = (ImageView) findViewById(android.R.id.icon);
        sideNavigationView = (SideNavigationView) findViewById(R.id.side_navigation_view);
        sideNavigationView.setMenuItems(R.menu.side_navigation_menu);
        sideNavigationView.setMenuClickCallback(this);

        if (getIntent().hasExtra(EXTRA_TITLE)) {
            String title = getIntent().getStringExtra(EXTRA_TITLE);
            int resId = getIntent().getIntExtra(EXTRA_RESOURCE_ID, 0);
            setTitle(title);
            icon.setImageResource(resId);
        }

        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                sideNavigationView.toggleMenu();
                break;
            default:
                return super.onOptionsItemSelected(item);
        }
        return true;
    }

    @Override
    public void onSideNavigationItemClick(int itemId) {
        switch (itemId) {
            case R.id.side_navigation_menu_item1:
                invokeActivity(getString(R.string.title1), R.drawable.ic_android1);
                break;

            case R.id.side_navigation_menu_item2:
                invokeActivity(getString(R.string.title2), R.drawable.ic_android2);
                break;

            case R.id.side_navigation_menu_item3:
                invokeActivity(getString(R.string.title3), R.drawable.ic_android3);
                break;

            case R.id.side_navigation_menu_item4:
                invokeActivity(getString(R.string.title4), R.drawable.ic_android4);
                break;

            case R.id.side_navigation_menu_item5:
                invokeActivity(getString(R.string.title5), R.drawable.ic_android5);
                break;

            default:
                return;
        }
        finish();
    }

    @Override
    public void onBackPressed() {
        // hide menu if it shown
        if (sideNavigationView.isShown()) {
            sideNavigationView.hideMenu();
        } else {
            super.onBackPressed();
        }
    }

    /**
     * Start activity from SideNavigation.
     * 
     * @param title title of Activity
     * @param resId resource if of background image
     */
    private void invokeActivity(String title, int resId) {
        Intent intent = new Intent(this, MainActivity.class);
        intent.putExtra(EXTRA_TITLE, title);
        intent.putExtra(EXTRA_RESOURCE_ID, resId);

        // all of the other activities on top of it will be closed and this
        // Intent will be delivered to the (now on top) old activity as a
        // new Intent.
        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);

        startActivity(intent);
        // no animation of transition
        overridePendingTransition(0, 0);
    }

}

File: sample.net/MainActivity.cs
using Android.App;
using Android.Content;
using Android.Views;
using Android.Widget;
using Android.OS;
using com.devspark.sidenavigation;

using Com.Actionbarsherlock.App;
using Com.Actionbarsherlock.View;
using Java.Lang;
using IMenu = global::Com.Actionbarsherlock.View.IMenu;
using IMenuItem = global::Com.Actionbarsherlock.View.IMenuItem;
using MenuItem = global::Com.Actionbarsherlock.View.MenuItem;
using ISubMenu = global::Com.Actionbarsherlock.View.ISubMenu;

/**
 * 
 * @author hram
 *
 */
namespace sample.net
{
    [Activity(Label = "@string/app_name", MainLauncher = true, Icon = "@drawable/ic_launcher")]
    public class MainActivity : SherlockFragmentActivity, ISideNavigationCallback
    {
        public static string EXTRA_TITLE = "com.devspark.sidenavigation.sample.extra.MTGOBJECT";
	    public static string EXTRA_RESOURCE_ID = "com.devspark.sidenavigation.sample.extra.RESOURCE_ID";

        private ImageView icon;
        private SideNavigationView sideNavigationView;

        protected override void OnCreate(Bundle bundle)
        {
            base.OnCreate(bundle);

            SetContentView(Resource.Layout.activity_main);
            icon = FindViewById<ImageView>(Android.Resource.Id.Icon);
            sideNavigationView = FindViewById<SideNavigationView>(Resource.Id.side_navigation_view);
            sideNavigationView.setMenuItems(Resource.Menu.side_navigation_menu);
            sideNavigationView.setMenuClickCallback(this);

            if (Intent.HasExtra(EXTRA_TITLE))
            {
                string title = Intent.GetStringExtra(EXTRA_TITLE);
                int resId = Intent.GetIntExtra(EXTRA_RESOURCE_ID, 0);
                Title = title;
                icon.SetImageResource(resId);
            }

            SupportActionBar.SetDisplayHomeAsUpEnabled(true);
        }

        public override bool OnOptionsItemSelected(IMenuItem item)
        {
            switch (item.ItemId)
            {
		    case Android.Resource.Id.Home:
			    sideNavigationView.toggleMenu();
			    break;
		    default:
			    return base.OnOptionsItemSelected(item);
		    }
		    return true;
	    }

        public void onSideNavigationItemClick(int itemId)
        {
		    switch (itemId) {
		    case Resource.Id.side_navigation_menu_item1:
			    invokeActivity(GetString(Resource.String.title1), Resource.Drawable.ic_android1);
			    break;

		    case Resource.Id.side_navigation_menu_item2:
			    invokeActivity(GetString(Resource.String.title2), Resource.Drawable.ic_android2);
			    break;

		    case Resource.Id.side_navigation_menu_item3:
			    invokeActivity(GetString(Resource.String.title3), Resource.Drawable.ic_android3);
			    break;

		    case Resource.Id.side_navigation_menu_item4:
			    invokeActivity(GetString(Resource.String.title4), Resource.Drawable.ic_android4);
			    break;

		    case Resource.Id.side_navigation_menu_item5:
			    invokeActivity(GetString(Resource.String.title5), Resource.Drawable.ic_android5);
			    break;

		    default:
			    return;
		    }
		    Finish();
	    }

        public override void OnBackPressed()
        {
		    // hide menu if it shown
		    if (sideNavigationView.Visibility == ViewStates.Gone) {
			    sideNavigationView.hideMenu();
		    } else {
                base.OnBackPressed();
		    }
	    }

	    /**
	     * Start activity from SideNavigation.
	     * 
	     * @param title
	     *            title of Activity
	     * @param resId
	     *            resource if of background image
	     */
	    private void invokeActivity(string title, int resId) {
		    Intent intent = new Intent(this, typeof (MainActivity));
		    intent.PutExtra(EXTRA_TITLE, title);
		    intent.PutExtra(EXTRA_RESOURCE_ID, resId);

		    // all of the other activities on top of it will be closed and this
		    // Intent will be delivered to the (now on top) old activity as a
		    // new Intent.
            intent.AddFlags(ActivityFlags.ClearTop);

		    StartActivity(intent);
		    // no animation of transition
		    OverridePendingTransition(0, 0);
	    }
    }
}

